lib/elixir/lib/macro.ex

```diff
@@ -2642,8 +2642,23 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:if, _meta, [condition_ast, _clauses]} = ast, env) do
+    case Macro.Env.lookup_import(env, {:if, 2}) do
+      [macro: Kernel] ->
+        quote do
+          condition_result = unquote(condition_ast)
+          result = unquote(ast)
```

Comment:
Since `ast` contains `condition_ast`, you are unquoting it twice and therefore I think it will call your condition twice.
Should probably be something like
```suggestion
          condition_result = unquote(condition_ast)
          result = if condition_result, unquote(clauses)
```

Reply:
Yes, confirmed it fixed it (before/after):

<img width="346" alt="Screenshot 2024-05-29 at 8 23 59" src="https://github.com/elixir-lang/elixir/assets/11598866/571896fd-953d-405b-8839-bcd57f6c55d6">

<img width="341" alt="Screenshot 2024-05-29 at 8 26 30" src="https://github.com/elixir-lang/elixir/assets/11598866/211c49c3-2e5d-448c-a5fd-181d5c2525cc">

Reply:
Oh sorry, using `if` created a bootstrapping issue that failed the CI.